python/dolfinx/fem/element.py

```diff
@@ -160,3 +162,190 @@ def _(e: basix.finite_element.FiniteElement):
         return CoordinateElement(_cpp.fem.CoordinateElement_float32(e._e))
     except TypeError:
         return CoordinateElement(_cpp.fem.CoordinateElement_float64(e._e))
+
+
+class FiniteElement:
+    _cpp_object: typing.Union[_cpp.fem.FiniteElement_float32, _cpp.fem.FiniteElement_float64]
+
+    def __init__(
+        self,
+        cpp_object: typing.Union[_cpp.fem.FiniteElement_float32, _cpp.fem.FiniteElement_float64],
+    ):
+        """Creates a Python wrapper for the exported finite element class.
+
+        Note:
+            Do not use this constructor directly. Instead use :func:`finiteelement`.
+
+        Args:
+            The underlying cpp instance that this object will wrap.
+        """
+        self._cpp_object = cpp_object
+
+    def __eq__(self, other):
+        return self._cpp_object == other._cpp_object
+
+    @property
+    def dtype(self) -> np.dtype:
+        """Geometry type of the Mesh that the FunctionSpace is defined on."""
+        return self._cpp_object.dtype
+
+    @property
+    def basix_element(self) -> basix.finite_element.FiniteElement:
+        """Return underlying Basix C++ element (if it exists).
+
+        Raises:
+            Runtime error if Basix element does not exist.
+        """
+        return self._cpp_object.basix_element
+
+    @property
+    def num_sub_elements(self) -> int:
+        """Number of sub elements (for a mixed or blocked element)."""
+        return self._cpp_object.num_sub_elements
+
+    @property
+    def value_shape(self) -> npt.NDArray[np.integer]:
+        """Value shape of the finite element field.
+
+        The value shape describes the shape of the finite element field, e.g. `{}` for a scalar,
+        `{2}` for a vector in 2D, `{3, 3}` for a rank-2 tensor in 3D, etc.
+        """
+        return self._cpp_object.value_shape
+
+    @property
+    def interpolation_points(self) -> npt.NDArray[np.floating]:
+        """Points on the reference cell at which an expression needs to be evaluated in order to
+        interpolate the expression in the finite element space.
+
+        Interpolation point coordinates on the reference cell, returning the (0) coordinates data
+        (row-major) storage and (1) the shape `(num_points, tdim)`.
```

Comment:
Python docs use reStructured text, so need double ticks for typewriting fonts, e.g. ``(num_points, tdim)``.

Return type descriptions is wrong -  returns an ndarray, not a flat array + shape.